src/app/loadout/loadout-edit/useEquipDropTargets.tsx

```diff
@@ -0,0 +1,51 @@
+import { DimItem } from 'app/inventory/item-types';
+import { singularBucketHashes } from 'app/loadout-drawer/loadout-utils';
+import { itemCanBeInLoadout } from 'app/utils/item-utils';
+import { DestinyClass } from 'bungie-api-ts/destiny2';
+import { TargetType } from 'dnd-core';
+import { DropTargetHookSpec, useDrop } from 'react-dnd';
+
+/**
+ * Hook to setup dnd drops for loadout editing.
+ *
+ * The accept param is used in a dependency array so ensure to memo it.
+ */
+export function useEquipDropTargets(accept: TargetType, classType: DestinyClass) {
+  const dropSpec =
+    (type: 'equipped' | 'unequipped') =>
+    (): DropTargetHookSpec<
+      DimItem,
+      { equipped: boolean },
+      { isOver: boolean; canDrop: boolean }
+    > => ({
+      accept,
+      drop: () => ({ equipped: type === 'equipped' }),
+      canDrop: (i) =>
+        itemCanBeInLoadout(i) &&
+        (i.classType === DestinyClass.Unknown || classType === i.classType) &&
+        (type === 'equipped' || !singularBucketHashes.includes(i.bucket.hash)),
+      collect: (monitor) => ({
+        isOver: monitor.isOver() && monitor.canDrop(),
+        canDrop: monitor.canDrop(),
+      }),
+    });
+
+  const [{ isOver: isOverEquipped, canDrop: canDropEquipped }, equippedRef] = useDrop(
+    dropSpec('equipped'),
```

Comment:
Do the dropSpecs need to be memoized?

Reply:
Not sure, will look into it.

Reply:
No it doesn't, see https://react-dnd.github.io/react-dnd/docs/api/use-drop. Basically the dependency array we are passing in as the second param is used in their own memoization.

Reply:
From my understanding, while this will get recreated it won't actually change the implementation returned by `useDrop` which should be fine.

Reply:
Happy to wrap it in a `useMemo` if you prefer though, no strong opinion.